content/specs/auth.md

```diff
@@ -0,0 +1,390 @@
+
+- `redirect_uri` (string, multiple allowed, optional): allows declaring a local redirect/callback URL. including a callback URL). Otherwise `http://127.0.0.1/` and `http://[::1]/` will be used.
+- `scope` (string, space-separated, multiple allowed, optional): default is `atproto`
+
+The other parameters in the virtual client metadata document will be:
+
+- `client_id` (string): the exact `client_id` (URL) used to generate the virtual document
+- `client_name` (string): a value chosen by the Authorization Server (e.g. "Loopback client")
+- `response_types` (array of strings): includes `code`
+- `grant_types` (array of strings): `authorization_code` and `refresh_token`
+- `token_endpoint_auth_method`: `none`
+- `application_type`: `native`
+- `dpop_bound_access_tokens`: `true`
+
+Note that this works as a public client, not a confidential client.
+
+### Identity Authentication
+
+As mentioned in the introduction, OAuth 2.0 generally provides only Authorization (`authz`), and additional standards like OpenID/OIDC are used for Authentication (`authn`). The atproto profile of OAuth requires authentication of account identity and supports the use case of simple identity authentication without additional resource access authorization.
+
+In atproto, account identity is anchored in the account DID, which is the permanent, globally unique, publicly resolvable identifier for the account. The DID resolves to a DID document which indicates the current PDS host location for the account. That PDS (combined with an optional entryway) is the authorization authority for the account, and is the OAuth Authorization Server for that account. When speaking to any Authorization Server, it is critical (mandatory) for clients to confirm that it is actually the authoritative server for the account in question, which means independently resolving the account identity (by DID) and confirming that the Authorization Server matches. It is also critical (mandatory) to confirm at the end of an authorization flow that the Authorization Server actually granted tokens for the expected account.
+
+Clients can start an auth flow in one of two ways:
+
+- starting with a public account identifier, provided by the user: handle or DID
+- starting with a server hostname, provided by the user: PDS or entryway, mapping to either Resource Server and/or Authorization Server
+
+Some use cases for starting with a server instead of an account identifier is if the user does not remember their full account handle, or only knows their account email.  Another is authentication when a user’s handle is broken. They still need to know their hosting provider in these situation.
```

Comment:
```suggestion
One use case for starting with a server instead of an account identifier is when the user does not remember their full account handle or only knows their account email.  Another is for authentication when a user’s handle is broken. The user will still need to know their hosting provider in these situation.
```